sqlmesh/core/plan/definition.py

```diff
@@ -255,12 +225,21 @@ def missing_intervals(self) -> t.List[SnapshotIntervals]:
     @property
     def snapshots(self) -> t.List[Snapshot]:
         """Gets all the snapshots in the plan/environment."""
-        return list(self.context_diff.snapshots.values())
+        return self._snapshots
+
+    @property
+    def snapshot_mapping(self) -> t.Dict[str, Snapshot]:
```

Comment:
[Nit] Can we make this "private" (underscored)? I think `snapshot_mapping` is not a commonly adopter terminology around the code base so I think it makes more sense to keep it as an implementation detail of the plan definition.

Reply:
Btw, why even make it a property method? Can't we just do:
```
self._snapshot_mapping = self.context_diff.snapshots
```
in the constructor? What's the point of re-creating the dict all the time?

Reply:
The snapshots in the plan do not equal the snapshots in context diff. Context diff is all snapshots and plan is a subset of that once it takes out snapshots that are `ignored`.